src/App.tsx

```diff
@@ -15,37 +18,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SetMode {
+  alphabetically = 'alphabetically',
+  length = 'length',
+  reset = 'reset',
+}
+
+type PreparedGoods = Array<string>;
+
+function getPreparedGoods(
+  goods: PreparedGoods,
+  mode: SetMode,
+  reverse: boolean,
+): PreparedGoods {
+  const goodsToSort = [...goods];
+
+  goodsToSort.sort((good1: string, good2: string): number => {
+    switch (mode) {
+      case SetMode.alphabetically:
+        return good1.localeCompare(good2);
+
+      case SetMode.length:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (reverse) {
+    goodsToSort.reverse();
+  }
+
+  return goodsToSort;
+}
+
 export const App: React.FC = () => {
+  const [actualSortMode, setActualSortMode] = useState(SetMode.reset);
+  const [listReverse, setListReverse] = useState(false);
+  const goodsToRender = getPreparedGoods(
+    goodsFromServer,
+    actualSortMode,
+    listReverse,
+  );
+
+  const setRevers = () => {
+    setActualSortMode(SetMode.reset);
+    setListReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': !(actualSortMode === SetMode.alphabetically),
+          })}
+          onClick={() => setActualSortMode(SetMode.alphabetically)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': !(actualSortMode === SetMode.length),
+          })}
+          onClick={() => setActualSortMode(SetMode.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !listReverse,
+          })}
+          onClick={() => setListReverse(currrent => !currrent)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(!(actualSortMode === SetMode.reset) || listReverse) && (
```

Comment:
Move this condition to a variable, so it will be easier to read and the JSX part will be less overloaded